File: src/test/java/runtime/parser/TestObjectReader.java
package runtime.parser;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class TestObjectReader {

	@Test
	public void test_p1e1() {
		ObjectReader input = new ObjectReader("nums = [2,7,11,15], target = 9");
		assertThat(input.read_param(int[].class))
				.isEqualTo(new int[]{2, 7, 11, 15});
		assertThat(input.read_param(int.class))
				.isEqualTo(9);
		assertThat(input.eof())
				.isTrue();

		ObjectReader output = new ObjectReader("[0,1]");
		assertThat(output.read_return(int[].class))
				.isEqualTo(new int[]{0, 1});
		assertThat(output.eof())
				.isTrue();
	}

}

File: leetcode-scala/src/test/scala/runtime/parser/test_object_reader.scala
package runtime.parser

import org.scalatest.funsuite.AnyFunSuite

class test_object_reader extends AnyFunSuite {

	test("p1e1") {
		val input = new ObjectReader("nums = [2,7,11,15], target = 9");
		assert(input.read_param(classOf[Array[Int]]) === Array(2, 7, 11, 15))
		assert(input.read_param(classOf[Int]) === 9)
		assert(input.eof)

		val output = new ObjectReader("[0,1]");
		assert(output.read_return(classOf[Array[Int]]) === Array(0, 1))
		assert(output.eof)
	}

}
